packages/blade/src/components/Input/BaseInput/StyledBaseInput.native.tsx

```diff
@@ -1,49 +1,39 @@
 import type { ReactElement } from 'react';
 import { useState } from 'react';
 import styled from 'styled-components/native';
-import type { BaseInputProps } from './BaseInput';
+import type { StyledBaseInputProps } from './StyledBaseInput.d';
+
 import getBaseInputStyles from './getBaseInputStyles';
 import type { Theme } from '~components/BladeProvider';
 
-export const StyledNativeBaseInput = styled.TextInput(
-  (props: BaseInputProps & { isFocussed: boolean; theme: Theme }) => ({
-    ...getBaseInputStyles({ theme: props.theme }),
-    backgroundColor: props.isFocussed
-      ? props.theme.colors.brand.primary[300]
-      : props.theme.colors.brand.gray[200],
-    borderBottomColor: props.isFocussed
-      ? props.theme.colors.brand.primary[500]
-      : props.theme.colors.brand.gray[400],
+const StyledNativeBaseInput = styled.TextInput<
+  StyledBaseInputProps & { isFocused: boolean; theme: Theme }
+>((props) => ({
+  ...getBaseInputStyles({
+    theme: props.theme,
+    isFocused: props.isFocused,
+    isDisabled: !props.editable,
+    validationState: props.validationState,
   }),
-);
+}));
 
 export const StyledBaseInput = ({
-  label,
-  labelPosition = 'top',
-  placeholder,
-  type = 'text',
-  defaultValue,
   name,
-  onChange,
-}: BaseInputProps): ReactElement => {
-  const [isFocussed, setIsFocussed] = useState(false);
-  console.log({
-    label,
-    labelPosition,
-    onChange,
-  });
+  isRequired,
+  isDisabled,
+  handleOnChange,
+  ...props
+}: StyledBaseInputProps): ReactElement => {
+  const [isFocused, setisFocused] = useState(false);
```

Comment:
We can utilize our useInteraction hook. 

```ts
  const { currentInteraction, onBlur, onFocus } = useInteraction();
  const isFocused = currentInteraction === 'focus';
```

Reply:
let me check that

Reply:
what is the benefit of this useInteraction?